test/modules/MetricsValidator/src/Program.cs

```diff
@@ -51,30 +51,46 @@ static async Task<int> MainAsync()
                     Logger.LogInformation("Open Async");
                     await moduleClient.OpenAsync();
 
+                    SemaphoreSlim directMethodProcessingLock = new SemaphoreSlim(1, 1);
+
                     Logger.LogInformation("Set method handler");
                     await moduleClient.SetMethodHandlerAsync(
                         "ValidateMetrics",
                         async (MethodRequest methodRequest, object _) =>
                         {
-                            Logger.LogInformation("Validating metrics");
-
-                            TestReporter testReporter = new TestReporter("Metrics Validation");
-                            List<TestBase> tests = new List<TestBase>
+                            Logger.LogInformation("Received method call to validate metrics");
+                            try
                             {
-                                new ValidateMessages(testReporter, scraper, moduleClient, transportType),
-                                new ValidateDocumentedMetrics(testReporter, scraper, moduleClient),
-                                // new ValidateHostRanges(testReporter, scraper, moduleClient),
-                            };
+                                await directMethodProcessingLock.WaitAsync();
 
-                            using (testReporter.MeasureDuration())
-                            {
-                                await Task.WhenAll(tests.Select(test => test.Start(cts.Token)));
-                            }
+                                // Delay to give buffer between potentially repeated direct method calls
+                                await Task.Delay(TimeSpan.FromSeconds(5));
```

Comment:
How did you find out that 5 seconds is good enough to handle the dup calls? how many are we getting currently in span of 5 secs?

Reply:
This delay is probably not required. The real synchronization is the lock above it. I just wanted some delay buffer for sanity. It's possible this isn't a good practice, but seemed fine to me since the real synchronization is in the locking.